crates/ruff_linter/src/rules/flake8_simplify/rules/split_static_string.rs

```diff
@@ -160,14 +165,42 @@ fn split_default(str_value: &StringLiteralValue, max_split: i32) -> Option<Expr>
             None
         }
         Ordering::Equal => {
-            let list_items: Vec<&str> = vec![str_value.to_str()];
-            Some(construct_replacement(
-                &list_items,
-                str_value.first_literal_flags(),
-            ))
+            // Behavior for maxsplit = 0 when sep is None:
+            // - If the string is empty or all whitespace, result is [].
+            // - Otherwise:
+            //   - " x ".split(maxsplit=0)  -> ['x ']
+            //   - " x ".rsplit(maxsplit=0) -> [' x']
+            //   - "".split(maxsplit=0) -> []
+            //   - " ".split(maxsplit=0) -> []
+            if string_val
+                .trim_matches(|c: char| c.is_whitespace())
```

Comment:
This is actually the exact implementation of [`trim`](https://doc.rust-lang.org/src/core/str/mod.rs.html#2125) 🙂 
```suggestion
                .trim()
```